### ISSUE TITLE
QLC+ on OnePlus6/SDM845 with PostmarketOS

### ISSUE THREAD

**Issue by ubl8** [mcallegari/qlcplus#1385]
**Describe the bug**
Hi, I am using my OnePlus6 with PostmarketOS installed as my daily driver.
I thought it would be really cool to use qlcplus on this device connected with a wireless ArtNet controller, for smal quick setups.

So I tried to build qlcplus on my OnePlus6 and it actually completed the build after I found the correct dependencies to install.

I don't know which one are the correct dependencies. 
So I installed some I found and some from the build log once it failed.

The installed dependencies look like the following:
```
sudo apk add build-base install-makedepends qt5-qtbase qt5-qtbase-dev qt5-qtquickcontrols qt5-qtquickcontrols2 qt5-qtquickcontrols2-dev qtchooser qt5-qtmultimedia qt5-qtmultimedia-dev qt5-qtscript qt5-qtscript-dev qt5-qttools qt5-qttools-dev qt5-qtdeclarative-dev alsa-lib-dev qt5-qtsvg-dev qt5-qtsvg qt5-qt3d qt5-qt3d-dev eudev-libs eudev-dev libftdi1-dev libftdi1
```
Not shure if this are all needed or if there are still some missing.
But at least the build completes, both for QLC+4 and for QLC+5.

Unfortunatly when I try to run qlcplus I get a `segmentation fault` and it exits.

**To Reproduce**
1. Obviously you need PostmarketOS (I build with an ARM device)
2. Install the dependencies (check above)
3. clone the repository to the device and enter the folder
4. build with `qmake && make` then install with `sudo make install`
5. execute the binary `qlcplus`

**Expected behavior**
QLC+ should run as normal and show the window.

But it fails with a `segmentation fault`.

**Desktop (please complete the following information):**
 - OS: Linux PostmarketOS (ARM) - wm: Phosh (wayland) 
          `Linux opmos 6.1.0-sdm845 #1-postmarketos-qcom-sdm845 SMP PREEMPT Fri Dec 16 10:59:11 UTC  aarch64 Linux`
 - QLC+ Version: git main branch
- hardware setup: OnePlus6

**Additional context**
The same errors appear as above when building QLC+5.
QLC+5 also won't create a desktop shortcut when Installed (QLC+4 does).

I can run the Fixture Editor without any problems.

I might try to build for Alma Linux (PostmarketOS is based on Alma Linux) with x86_64 desktop when I have time, to see if it has the same issue.

**Comment by ubl8** [comment]
Because it runs wayland I tried to run with different options for `QT_QPA_PLATFORM` and `XDG_SESSION_TYPE` but nothing changed so far.

I also used different options like `-f -g -m -n` but again nothing changed.

**Comment by mcallegari** [comment]
With the information provided, I can't help here.
This requires a proper debug session and I can't assist you on that

**Comment by ubl8** [comment]
I just spend a few more minutes on the Issue and made some more progress.

First of all I compiled QLC+5 with debugging enabled 
`qmake CONFIG+=qmlui CONFIG+=declarative_debug CONFIG+=qml_debug`

When running `qlcplus-qml` it fails with the following output
```
QWaylandGLContext: failed to create EGLContext, error=3009
Failed to create OpenGL context for format QSurfaceFormat(version 3.3, options QFlags<QSurfaceFormat::FormatOption>(), depthBufferSize 24, redBufferSize -1, greenBufferSize -1, blueBufferSize -1, alphaBufferSize -1, stencilBufferSize 8, samples -1, swapBehavior QSurfaceFormat::DoubleBuffer, swapInterval 1, colorSpace QSurfaceFormat::DefaultColorSpace, profile  QSurfaceFormat::CoreProfile)
```

On the web other Qt applications suffer the same problem especially on wayland.
So I tried executing with `QT_QPA_PLATFORM=xcbqlcplus-qml` but it fails again with `Failed to create Context: 3009`.

After some more research I found some one who got kde plasma only running with software rendering by passing `QMLSCENE_DEVICE=softwarecontext`.

And indeed launching the application with `QMLSCENE_DEVICE=softwarecontext qlcplus-qml` runns without failure.
And it is quite usable with software rendering.

This might be because of some graphics issue on my device.

I am not quite familiar with Qt5 and QML but this might help some.

**Comment by ubl8** [comment]
Okay, it seems to be the 3d preview is the main issue.

I installed the following packages (I can't prove if these are necessary, have to make a clean install first).
`sudo apk add wayland-libs-egl mesa-utils mesa-dev mesa-egl`

After that I compiled QLC+4 without cleaning the build files from QLC+5.
And QLC+4 runs without any further extra parameter.

I also found out that now QLC+5 runs with the 3d preview disabled.
No software rendering necessary.
